MoreCyclopsUpgrades/Managers/UpgradeManager.cs

```diff
@@ -103,35 +60,95 @@ internal bool Initialize(CyclopsManager manager)
 
             this.Manager = manager;
 
-            SimpleUpgradeActions.Add(TechType.CyclopsShieldModule, EnabledShield);
-            SimpleUpgradeActions.Add(TechType.CyclopsSonarModule, EnableSonar);
-            SimpleUpgradeActions.Add(TechType.CyclopsSeamothRepairModule, EnableRepairDock);
-            SimpleUpgradeActions.Add(TechType.CyclopsDecoyModule, EnableExtraDecoySlots);
-            SimpleUpgradeActions.Add(TechType.CyclopsFireSuppressionModule, EnableFireSuppressionSystem);
-            SimpleUpgradeActions.Add(TechType.CyclopsThermalReactorModule, AddThermalModule);
-            SimpleUpgradeActions.Add(TechType.PowerUpgradeModule, AddPowerMk1Module);
+            SetupPowerManagerUpgrades();
 
-            SimpleUpgradeActions.Add(CyclopsModule.SolarChargerID, AddSolarModule);
-            SimpleUpgradeActions.Add(CyclopsModule.SpeedBoosterModuleID, AddSpeedModule);
-            SimpleUpgradeActions.Add(CyclopsModule.PowerUpgradeMk2ID, AddPowerMk2Module);
-            SimpleUpgradeActions.Add(CyclopsModule.PowerUpgradeMk3ID, AddPowerMk3Module);
-            SimpleUpgradeActions.Add(CyclopsModule.BioReactorBoosterID, AddBioBooster);
+            RegisterUpgradeHandlers();
 
-            SlotBoundUpgradeActions.Add(CyclopsModule.SolarChargerMk2ID, AddSolarMk2Module);
-            SlotBoundUpgradeActions.Add(CyclopsModule.ThermalChargerMk2ID, AddThermalMk2Module);
-            SlotBoundUpgradeActions.Add(CyclopsModule.NuclearChargerID, AddNuclearModule);
+            SyncUpgradeConsoles();
 
-            ChargingModules.Add(CyclopsModule.SolarChargerID);
-            ChargingModules.Add(CyclopsModule.SolarChargerMk2ID);
-            ChargingModules.Add(TechType.CyclopsThermalReactorModule);
-            ChargingModules.Add(CyclopsModule.ThermalChargerMk2ID);
-            ChargingModules.Add(CyclopsModule.NuclearChargerID);
+            return true;
+        }
 
-            holographicHUD = this.Cyclops.GetComponent<CyclopsHolographicHUD>();
+        private void SetupPowerManagerUpgrades()
```

Comment:
Because this is executed from a non-static context, and is adding methods to a static handler,  
we should be sure that we aren't going to run into duplicate key exceptions.  
This would come into play if there is ever more than one Cyclops in the game.

An `UpgradeHandler` could have a field informing if the delegate that create it should be de-registered.  
We might instead see if there is another way to collect the delegate methods and confirm if a new instance for a known upgrade is being provided.

Consider the options and  find something that makes sense and won't be a pain for others to use.